internal/namespaces/instance/v1/custom_server_type.go

```diff
@@ -33,30 +47,39 @@ func serverTypeListBuilder(c *core.Command) *core.Command {
 		"X64-60GB":  {},
 	}
 
-	originalRun := c.Run
-
 	c.Run = func(ctx context.Context, argsI interface{}) (interface{}, error) {
 		type customServerType struct {
-			Name            string        `json:"name"`
-			MonthlyPrice    *scw.Money    `json:"monthly_price"`
-			HourlyPrice     *scw.Money    `json:"hourly_price"`
-			LocalVolumeSize scw.Size      `json:"local_volume_size"`
-			CPU             uint32        `json:"cpu"`
-			GPU             *uint64       `json:"gpu"`
-			RAM             scw.Size      `json:"ram"`
-			Arch            instance.Arch `json:"arch"`
+			Name            string                           `json:"name"`
+			MonthlyPrice    *scw.Money                       `json:"monthly_price"`
+			HourlyPrice     *scw.Money                       `json:"hourly_price"`
+			LocalVolumeSize scw.Size                         `json:"local_volume_size"`
+			CPU             uint32                           `json:"cpu"`
+			GPU             *uint64                          `json:"gpu"`
+			RAM             scw.Size                         `json:"ram"`
+			Arch            instance.Arch                    `json:"arch"`
+			Availability    instance.ServerTypesAvailability `json:"availability"`
 		}
 
-		originalRes, err := originalRun(ctx, argsI)
+		api := instance.NewAPI(core.ExtractClient(ctx))
+
+		// Get server types.
+		request := argsI.(*instance.ListServersTypesRequest)
+		listServersTypesResponse, err := api.ListServersTypes(request)
 		if err != nil {
 			return nil, err
 		}
-
-		listServersTypesResponse := originalRes.(*instance.ListServersTypesResponse)
 		serverTypes := []*customServerType(nil)
 
-		for name, serverType := range listServersTypesResponse.Servers {
+		// Get server availabilities.
+		serverAvailabilities := make(map[string]instance.ServerTypesAvailability)
+		availabilitiesResponse, err := api.GetServerTypesAvailability(&instance.GetServerTypesAvailabilityRequest{})
+		if err != nil {
+			// Do nothing and show degraded results.
```

Comment:
We should be really careful with this type of optimization. While it's a excellent idea for human display it could be a source of many strange bugs when doing scripting and automation as the command will not return an error but will not contain the correct information? 

Can we consider this case will almost never happen and that user experience will almost never be affected by this use-case and at the same time guarantee automation?

Reply:
Good catch, what is your point? Do you prefer to return an error here? I agree.

Reply:
yes let's return an error